starboard/BUILD.gn

```diff
@@ -177,24 +181,8 @@ source_set("starboard_headers_only") {
 }
 
 if (current_toolchain == starboard_toolchain) {
-  target(starboard_target_type, "starboard_platform_group") {
```

Comment:
Was there anything functionally wrong with the template?
If not, let's not change it here. Even if the refactoring is a noop good cleanup it should be in separate PR.

Reply:
Extra configs need to be passed to this target for the build to work, it needs to be done in this PR